tests/run-make/alloc-no-oom-handling/rmake.rs

```diff
@@ -0,0 +1,18 @@
+// ignore-tidy-linelength
+
+extern crate run_make_support;
+
+use run_make_support::rustc;
+
+fn main() {
+    rustc()
+        .arg("--edition")
+        .arg("2021")
+        .arg("-Dwarnings")
+        .arg("--crate-type")
+        .arg("rlib")
+        .arg("../../../library/alloc/src/lib.rs")
```

Comment:
You may need to properly construct this path via `PathBuf`, I think this will fail on Windows because Windows uses `\` for path separators. You could try it out by editing PR CI to run the `x86_64-msvc` target and running `./x test tests/run-make --stage 1` or similar.

Reply:
@jieyouxu thanks for reviewing, I will edit the workflow to test it on windows

Reply:
Actually maybe that's not true *for input file argument passed to rustc*. I checked, the compiler converts this input file path string into a `PathBuf` super early on. However, for consistency with non-rustc tools that does not do this, maybe it's still better to construct this path via a `PathBuf` and explicitly convert that to a string to pass into `arg()`.

Reply:
The test is working on windows. I checked the arg function in Command has a conversion trait to OsStr for the arguments

Reply:
> The test is working on windows. I checked the arg function in Command has a conversion trait to OsStr for the argument

(to be pedantic, that itself doesn't mean `Command::arg` when passed a unix path string will work for arbitrary executables. It happens to work for `rustc` because `rustc` converts the unix path string to a `PathBuf` internally, and later on I presume it's been normalized/canonicalized upon actual usage)

Reply:
Sorry, I missed your previous comment. I will construct it explicilty using pathbuf. Can you tell me how I can check this "compiler converts this input file path string into a PathBuf super early on"?

Reply:
> Sorry, I missed your previous comment. I will construct it explicilty using pathbuf. Can you tell me how I can check this "compiler converts this input file path string into a PathBuf super early on"?

Note that I haven't followed the trail all the way to first usage, but here's the rough trail I looked at:

- In [`rustc_driver_impl`](https://github.com/rust-lang/rust/blob/9e73597e5a83e96e223c10b3810566c74a86a0c1/compiler/rustc_driver_impl/src/lib.rs), there's a `make_input` function which converts the input path argument into a `PathBuf`.
- This is used to build up a `rustc_session::config::Input`, which holds a `PathBuf` variant.
- This is used to build up a `rustc_interface::Config` which is then passed to `rustc_interface::run_compiler`.